flocker/control/test/test_persistence.py

```diff
@@ -249,3 +252,32 @@ def test_complex_keys(self):
                                manifestations={}, paths={},
                                devices={uuid4(): FilePath(b"/tmp")})
         self.assertEqual(node_state, wire_decode(wire_encode(node_state)))
+
+
+class ConfigurationMigrationTests(SynchronousTestCase):
+    """
+    Tests for ``_ConfigurationMigration``.
+    """
+    configurations_dir = FilePath(__file__).sibling('configurations')
+
+    def test_v0_v1_configuration(self):
+        """
+        A V0 JSON configuration blob is transformed to a V1 configuration
+        blob, with the result validating when loaded.
+        """
+        v0_json = self.configurations_dir.child(
+            'configuration_v0.json').getContent()
+        v1_json = migrate_configuration(0, 1, v0_json)
+        v1_config = wire_decode(v1_json)
+        self.assertEqual(v1_config, Deployment(nodes=frozenset()))
```

Comment:
Yeah, the problem I had here is that "version 0" configs don't really exist and are made up for the purposes of testing, plus the scope of this branch doesn't define a version 2 config (which will become a thing only when we do the branch that adds Leases to a Deployment) - so I couldn't think of any other example.